hw14/election.py

```diff
@@ -0,0 +1,321 @@
+    dictionary that maps a column name to a value in that column, as a string.
+    """
+    output = []
+    for row in csv.DictReader(open(path)):
+        output.append(row)
+    return output
+
+
+###############################################################################
+# Problem 1: State edges
+###############################################################################
+
+def row_to_edge(row):
+    """
+    Given an *ElectionDataRow* or *PollDataRow*, returns the
+    Democratic *Edge* in that *State*.
+    """
+    return float(row["Dem"]) - float(row["Rep"])
+
+
+def state_edges(election_result_rows):
+    """
+    Given a list of *ElectionDataRow*s, returns *StateEdge*s.
+    The input list has no duplicate *States*;
+    that is, each *State* is represented at most once in the input list.
+    """
+    d = {}
+    for row in election_result_rows:
+        state = row['State']
+        dem = row['Dem']
+        rep = row['Rep']
+
+        d[state] = float(dem) - float(rep)
+    return d
+
+###############################################################################
+# Problem 2: Find the most recent poll row
+###############################################################################
+
+
+def earlier_date(date1, date2):
+    """
+    Given two dates as strings (formatted like "Oct 06 2012"), returns True if
+    date1 is before date2.
+    """
+    return (time.strptime(date1, "%b %d %Y") < time.strptime(date2, "%b %d %Y"))
+
+
+def most_recent_poll_row(poll_rows, pollster, state):
+    """
+    Given a list of *PollDataRow*s, returns the most recent row with the
+    specified *Pollster* and *State*. If no such row exists, returns None.
+    """
+
+    returndata = 'None'
+    checkdate = 'Jan 01 1900'
+    for i in range(len(poll_rows)):
+        if state in poll_rows[i]['State'] and \
+           pollster in poll_rows[i]['Pollster']:
+            if earlier_date(checkdate, poll_rows[i]['Date']):
+                returndata = poll_rows[i]
+        else:
+            continue
+    if returndata == 'None':
+        return None
+    else:
+        return(returndata)
+
+
+###############################################################################
+# Problem 3: Pollster predictions
+###############################################################################
+
+def unique_column_values(rows, column_name):
+    """
+    Given a list of rows and the name of a column (a string),
+    returns a set containing all values in that column.
+    """
+
+    # Make a list of the values corresponding to column names
+    col_list = []
+    for row in rows:
+        col_list.append(row[(column_name)])
+
+    # Reformat/make the list unique using set
+    returndata = set(col_list)
+    return(returndata)
+
+
+def pollster_predictions(poll_rows):
+    """
+    Given a list of *PollDataRow*s, returns *PollsterPredictions*.
+    For a given pollster, uses only the most recent poll for a state.
+    """
+
+    d = {}
+    n = []
```

Comment:
Single character variable names can cause a lot of headaches. Rather use something semantically meaningful that expresses what the variable is bound to.